### PR TITLE
update udf constructors

### LINKED ISSUE #794: move user transformation/measurement constructors
The Analysis API only reads from the transformations/measurements modules. The user-defined trans/meas constructors should live in these modules, because then they'll get picked up in the Analysis API. It also makes intuitive sense, as these constructors don't take trans/meas as arguments.

### PR DESCRIPTION
**Pull request by Shoeboxam**
Closes #794

* Move `make_user_transformation` into `transformations`
	* so that it will appear in the HLAPI. It also isn't a combinator.
* Move `make_user_measurement` into `measurements`
 	* so that it will appear in the HLAPI. It also isn't a combinator.
* Move `make_user_postprocessor` into `core` and rename to `new_function`
    * consistent with `new_queryable` in #851 
    * naming was too specific before- will use it for other things than just postprocessors
* Reorder arguments to `make_user_transformation` and `make_user_measurement` to dom/met convention
    * so that you can do `>> then_user_transformation(DO, MO, func, map)`, etc

### CODE REVIEW

**Comment by Shoeboxam** [comment]
Current dependencies on/for this PR:
* main
  * **PR #884** <a href="https://app.graphite.dev/github/pr/opendp/opendp/884" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/opendp/opendp/884?utm_source=stack-comment).

File: rust/src/measurements/make_user_measurement/mod.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    privacy_map: CallbackFn,
+) -> Fallible<Measurement<AnyDomain, AnyObject, AnyMetric, AnyMeasure>> {
+    let _ = (
+        input_domain,
+        input_metric,
+        output_measure,
+        privacy_map,
+        function,
+    );
+    panic!("this signature only exists for code generation")
```
Comment (andrewvyrros) [inline rust/src/measurements/make_user_measurement/mod.rs:49]: Nice trick!

File: rust/src/measurements/make_user_measurement/mod.rs
Comment (andrewvyrros) [inline rust/src/measurements/make_user_measurement/mod.rs:1]: The module name feels a little odd. Wouldn't `user_defined` or `user_measurement` (or just `user`) fit the pattern better? (Same for transformation.)

File: rust/src/measurements/make_user_measurement/mod.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+#[bootstrap(
+    name = "make_user_measurement",
+    features("contrib", "honest-but-curious"),
+    arguments(
+        input_domain(hint = "Domain"),
+        input_metric(hint = "Metric"),
+        function(rust_type = "$pass_through(TO)"),
+        output_measure(hint = "Measure"),
```
Comment (andrewvyrros) [inline rust/src/measurements/make_user_measurement/mod.rs:18]: Minor: The order of these is swapped vs the function.

File: rust/src/measurements/make_user_measurement/mod.rs
Comment (Shoeboxam) [reply rust/src/measurements/make_user_measurement/mod.rs:1]: Anna/Pauline started naming modules after the constructor function, and I really like it.

Comment (andrewvyrros) [reply rust/src/measurements/make_user_measurement/mod.rs:1]: OK, if the crew is liking this convention, I'm cool with it.

**Review by andrewvyrros — APPROVED** [review]

**Review by andrewvyrros — APPROVED** [review]

**Pull request merged**